auth/template.yaml

```diff
@@ -2010,6 +2015,192 @@ Resources:
+    Condition: IsNotProduction
+    Type: AWS::WAFv2::WebACL
+    Properties:
+      Name: !Sub ${AWS::StackName}-shared-signals-waf
+      Scope: REGIONAL
+      Description: WAF for Shared Signals API
+      DefaultAction:
+        Allow: {}
+      VisibilityConfig:
+        SampledRequestsEnabled: true
+        CloudWatchMetricsEnabled: true
+        MetricName: !Join
+          - "-"
+          - - SharedSignalWAFMetric
+            - !Sub ${AWS::StackName}
+      Rules:
+        - Name: SharedSignalThrottlingRule
+          Priority: 1
+          Action:
+            Block:
+              CustomResponse:
+                ResponseCode: 429
+          Statement:
+            RateBasedStatement:
+              Limit: 100 # Evaluating 100 requests AgregateKeyType: IP
+              EvaluationWindowSec: 60 # 1 minute window
+              AggregateKeyType: IP
+          VisibilityConfig:
+            SampledRequestsEnabled: true
+            CloudWatchMetricsEnabled: true
+            MetricName: !Join
+              - "-"
+              - - SharedSignalWAFMetricRule
+                - !Sub ${AWS::StackName}
+        - Name: SharedSignalWAFIPAddressesRule
+          Priority: 2
+          Statement:
+            IPSetReferenceStatement:
+              Arn: !GetAtt IPAddressSet.Arn
+          Action:
+            Allow: {}
+          VisibilityConfig:
+            SampledRequestsEnabled: true
+            CloudWatchMetricsEnabled: true
+            MetricName: !Join
+              - "-"
+              - - SharedSignalWAFIPSetRule
+                - !Sub ${AWS::StackName}
+        - Name: SharedSignalAWSManagedBadInputsRuleSet
+          Priority: 3
+          Statement:
+            ManagedRuleGroupStatement:
+              VendorName: AWS
+              Name: AWSManagedRulesKnownBadInputsRuleSet
+          OverrideAction:
+            None: {}
+          VisibilityConfig:
+            SampledRequestsEnabled: true
+            CloudWatchMetricsEnabled: true
+            MetricName: !Join
+              - "-"
+              - - SharedSignalWAFBadInputsRule
+                - !Sub ${AWS::StackName}        
+      Tags:
+        - Key: Product
+          Value: GOV.UK
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: System
+          Value: Authentication   
+
+  SharedSignalWafLogGroup:
+    Condition: IsNotProduction
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupClass: STANDARD
+      # must starts with aws-waf-logs-
+      LogGroupName: !Sub aws-waf-logs-shared-signal-${AWS::StackName}
+      KmsKeyId: !GetAtt SharedSignalWafLogGroupKMSKey.Arn
+      RetentionInDays: 30
+      DataProtectionPolicy:
+        Name: CloudWatchLogs-PersonalInformation-Protection
+        Description: Protect basic types of sensitive data
+        Version: "2021-06-01"
+        Configuration:
+          CustomDataIdentifier:
+            - Name: IPAddresses
+              Regex: \b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b
```

Comment:
I think there's a data identifier already for IP addresses ([link](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/protect-sensitive-log-data-types-device.html))